test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -540,6 +554,45 @@
                  (initialize-master-ssl! settings "master" ca-settings 512)))
             (fs/copy copy path)))))))
 
+(deftest initialize-master-ssl!-test-with-keylength-settings
+  (let [tmp-confdir (fs/copy-dir confdir (ks/temp-dir))
+        settings (-> (testutils/master-settings tmp-confdir "master")
+                     (assoc :dns-alt-names "onefish,twofish"))
+        ca-settings (assoc (testutils/ca-settings (str tmp-confdir "/ssl/ca")) :keylength 768)]
```

Comment:
Nice job with these tests. I think we should use all of the default values from test util settings, and only specify the ones that we're testing. In this case, we're not testing the `:dns-alt-names` or the master certname, so I'd try to get rid of the `(assoc :dns-alt-names ...)` and the `"master"` parameter to `master-settings`.

We're going to be doing heavy refactoring of these tests at some point and keeping them as simple and straightforward as possible will go a long way.